packages/playwright/src/reporter.ts

```diff
@@ -44,19 +109,39 @@ class ReplayReporter implements Reporter {
   }
 
   onTestEnd(test: TestCase, result: TestResult) {
-    if (!["passed", "failed"].includes(result.status)) return;
+    const status = result.status;
+    if (status !== "passed" && status !== "failed") return;
+
+    const recs = listAllRecordings().filter((r) => {
+      if (
+        r.metadata["x-playwright"] &&
+        typeof r.metadata["x-playwright"] === "object"
+      ) {
+        return (r.metadata["x-playwright"] as any).id === this.getTestId(test);
+      }
+
+      return false;
+    });
 
-    const recs = listAllRecordings().filter(
-      (r) => r.metadata.testId === this.getTestId(test)
-    );
     if (recs.length > 0) {
       recs.forEach((rec) => {
         const metadata = {
           id: rec.id,
           kind: "addMetadata",
           metadata: {
             title: test.title,
-            testStatus: result.status,
+            ...testMetadata.init({
+              title: test.title,
+              result: status,
+              path: test.titlePath(),
+              run: {
+                id: this.baseId,
+                title: this.runTitle
+              },
+              // extract the relative path from titlePath() but fall back to the
+              // full path
+              file: test.titlePath()[2] || test.location.file,
```

Comment:
what is the shape of `titlePath()` here usually? Is it possible that we'll blow up trying to pull [2] from `undefined` or anything like that?